### PR TITLE
WIP Linux support

### PR DESCRIPTION
**Pull request by floooh**
Adds Linux support, ~~only thing I need still need to figure out is why the image is clipped (any ideas?).~~

Also, don't merge yet since I'd like to rebase this PR on top of the macOS PR after that is merged [DONE], and make the shader selection a bit cleaner [DONE].

- also fix a couple of tabs-vs-spaces issues in the CMakeLists.txt file (there were a couple of spaces in the Mac specific parts which I converted to tabs now).
- added a GLSL shader pair
- changed the shader selection to use the `SOKOL_*` backend defines instead of the 'raw' platform defines
- added Linux to the CI build matrix

NOTE to self:

- [ ] update README.md mentioning Linux and build instructions (dev-packages)

Screenshot_20240523_205510

### CODE REVIEW

**Comment by floooh** [comment]
Ah... Linux missing in CI, one sec.

Thanks for merging the other PRs, I'll continue with the Linux PR tomorrow then (resolve conflicts, code cleanup and figuring out why the image is clipped) :)

File: src/platform.c [hunk 1]
```diff
@@ -566,6 +582,21 @@ static const char* kSokolFragSource =
...
 "}\n";
+#else
+// GLSL
+"#version 410\n"
+"uniform sampler2D tex;\n"
+"in vec2 uv;\n"
+"out vec4 frag_color;\n"
+"void main() {\n"
+"  int x = int(uv.x * 400);\n"
+"  int y = int(uv.y * 200);\n"
```
Comment (aras-p) [inline src/platform.c]: Maybe this should be `* 240`, not `* 200` :) -- probably explains why the image is stretched/clipped vertically

Comment (floooh) [reply src/platform.c]: Damn! I need new glasses lol. Looks good now.

**Comment by floooh** [comment]
Ok, I think the Linux changes are ok now, I'll log off for today and look over the changes tomorrow again, and also check that I didn't break anything on Windows and Mac (before switching off the Draft flag).

**Comment by aras-p** [comment]
Too late! I just merged it.

**Comment by floooh** [comment]
Lol! Ok thanks :)

I guess it's fine, only the readme is missing Linux info. I can take care of that tomorrow. Have a good evening :)

**Comment by aras-p** [comment]
> only the readme is missing Linux info.

Too late for that too, since I just updated it :) Much thanks for the cleanups and the Linux support! And for Sokol in general.

**Pull request merged**